.github/workflows/pr-build.yml

```diff
@@ -38,31 +40,42 @@ jobs:
       - name: Build with Gradle
         run: ./gradlew build dockerBuildImages
 
-      - name: Unit test
-        run: ./gradlew jacocoTestReport
-
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
+  validate-helm-charts:
+    runs-on: ubuntu-20.04
+    container: 
+      image: hypertrace/helm-gcs-packager:0.3.0
+      credentials:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+     # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
         with:
-          name: codecov-line-smoothing
-          fail_ci_if_error: true
-          verbose: true
-          flags: unit
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
+          fetch-depth: 0
 
-      - name: Integration test
-        run: ./gradlew jacocoIntegrationTestReport
+      - name: validate charts
+        run: |
+          helm dependency update ./helm/
+          helm lint --strict ./helm/
+          helm template ./helm/
 
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v1
+  snyk-scan:
+    runs-on: ubuntu-20.04
+    steps:
+    # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
         with:
-          name: codecov-line-smoothing
-          fail_ci_if_error: true
-          verbose: true
-          flags: integration
-
-      - name: push docker image
-        if: github.event.pull_request.merged
-        run: ./gradlew dockerPushImages
-
+          ref: ${{github.event.pull_request.head.ref}}
+          repository: ${{github.event.pull_request.head.repo.full_name}}
```

Comment:
So I wrote the below comment, and then realized that the jobs are already sorted this way. So I think we should make the change in naming to reflect the actual distinction in the workflows, the triggers (currently the build job has several validation/test tasks):


> Should we make the `pull_request` workflow our default location for jobs, and limit `pull_request_target` to the small set of jobs that need access to secrets? The benefits I was thinking were that
> - The risk of accidental secret exposure decreases (because we're running less jobs in forked contexts)
> - The extra dance around checkout is needed on fewer jobs
> The drawbacks being-
> - The workflow would be sorted/named by trigger than than function (but the results are generally reported by job, not workflow)
>

Reply:
will change the name